File: EPI/13.1/swift/13.1.swift
public func anagrams(words: [String]) -> [[String]] {
    var sortedWordToAnagrams = [String: [String]]()

    for word in words {
        let sortedWord = String(word.characters.sort(<))

        if sortedWordToAnagrams[sortedWord] == nil {
            sortedWordToAnagrams[sortedWord] = [word]
        } else {
            sortedWordToAnagrams[sortedWord]!.append(word)
        }
    }

    let result = sortedWordToAnagrams.values.filter { $0.count > 1 }
    return [[String]](result)
}

File: EPI/13.1/typescript/13.1.ts
export default function groupAnagrams(words: string[]): string[][] {
    const sortedMap = new Map<string, string[]>();

    for (const word of words) {
        const sortedWord = sort(word);

        if (!sortedMap.has(sortedWord)) {
            sortedMap.set(sortedWord, []);
        }

        sortedMap.get(sortedWord)!.push(word);
    }

    return [...sortedMap.values()].filter(group => group.length > 1);
}

function sort(s: string): string {
    return [...s].sort().join('');
}
